docs/extensions/storage/proposedchanges.md

```diff
@@ -0,0 +1,185 @@
+
+Now we migrate the space `345` to another storage driver:
+- the storage registry contains a new entry for `instanceprefix$345` to send all resource ids for that space to the new storage provider
+- the new storage driver has to take into account the full storageid because the nodeid may only be unique per storage space.
+
+If we now have to fetch the path on the `/dav/meta/` endpoint:
+`/dav/meta/instanceprefix$345!1`
+`/dav/meta/instanceprefix$345!2`
+`/dav/meta/instanceprefix$345!3`
+
+This would work because the registry always sees `instanceprefix$345` as the storageid.
+
+Now if we use the fileids directly and leave out the numeric storageid:
+`<instanceprefix>!<fileid>`
+`instanceprefix!1`
+`instanceprefix!2`
+`instanceprefix!3`
+
+This is the current `<storageid>!<nodeid>` format.
+
+The reva storage registry contains a `instanceid` entry pointing to the storage provider with the owncloudsql driver.
+
+Resources can be looked up because the oc_filecache has a unique fileid over all storages.
+
+Now we again migrate the space `345` to another storage driver:
+- the storage registry contains a new entry for `instanceprefix!1` so the storage space root now points to the new storage provider
+- The registry needs to be aware of node ids to route properly. This is a no go. We don't want to keep a cache of *all* nodeids in the registry. Only the root nodes of spaces.
+- The new storage driver only has a nodeid which might collide with other nodeids from other storage spaces, eg when two instances are imported into one ocis instance. Although it would be possible to just set up two storage providers extra care would have to be taken to prevent nodeid collisions when importing a space.
+
+If we now have to fetch the path on the `/dav/meta/` endpoint:
+`/dav/meta/instanceprefix!1` would work because it is the root of a space
+`/dav/meta/instanceprefix!2` would cause the gateway to poll all storage providers because the registry has no way to determine the responsible storage provider
+`/dav/meta/instanceprefix!3` same
+
+The problem is that without a part in the storageid that allows differentiating storage spaces we cannot route them individually.
+
+Now, we could use the nodeid of the root of a storage space as the spaceid ... if it is a uuid. If it is numeric it needs a prefix to distinguish it from other spaces.
+`<space-root-uuid>!<fileid>` would be easy for the decomposedfs.
+eos might use numeric ids: `<eosprefix>$<space-root-fileid>!<fileid>`, but it needs a custom prefix to distinguish multiple eos instances.
+
+Furthermore, when migrating spaces between storage providers we want to stay collision free, which is why we should recommend uuids.
+
+All this has implications for the decomposedfs, because it needs to split the nodes per space to prevent them from colliding.
```

Comment:
General note: Maybe the aspects of migrations from oC10 storages should be moved to a separate doc. This is very hard to read tbh.